lab/lab-1.md

```diff
@@ -0,0 +1,209 @@
+
+## Deploy the frontend to Kubernetes
+
+Create a deployment manifest(`deployment_v1.yaml`) and apply it on the Kubernetes cluster using `kubectl`. Don't forget to specify some meaningful labels:
+
+- `app=frontend`
+- `version=v1`
+
+Put the deployment also in the namespace `demoapp`(The namespace was create in the `Deploy the backend to Kubernetes` section):
+
+```sh
+[centos@node-0 example-app-frontend]$ kubectl apply -f deployment_v1.yaml
+```
+
+If you need more information about how to create a kubernetes deployment manifest, check the docs [here](https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#creating-a-deployment). The deployment should run the **frontend** docker image in version `v1.0.0`.
+
+Don't forget to set the environment variable `API_URL` to a valid URL(`http://<MY_BACKEND_HOST>:<MY_BACKEND_PORT>`).
+How can you configure the environment for your deployment? Check out the docs about environment for containers [here](https://kubernetes.io/docs/tasks/inject-data-application/define-environment-variable-container/#define-an-environment-variable-for-a-container).
+
+## Expose the frontend service to the public internet
+
+Now we want to expose our frontend to world, so that everyone can use our application. Therefore we have to make the frontend publicly available. How can we do this? What options do we have and what is the easiest way to do this?
+
+Checkout the docs/intro about Kubernetes services [here](https://kubernetes.io/docs/tutorials/kubernetes-basics/expose/expose-intro/).
+Which type can we use?
+
+Choose an appropriate type, create a new manifest `service.yaml` and expose so the deployment on the cluster. Don't forget to put the resource into the correct namespace and use both labels(`app`, `version`) for the pod selection.
+
+```sh
+[centos@node-0 example-app-frontend]$ kubectl apply -f service.yaml
+```
+
+Afterwards open the frontend in your browser and check if the frontend shows the backend version `v1.0.0`. If not check the `API_URL`. Maybe there is something wrong with the settings.
+
+## Blue/Green deployment - Frontend
+
+### Build frontend v2.0.0
+
+We encountered that our users don't like the background color `blue`. Therfore we want to change it to `green`. The source code is already available:
+
+```sh
+[centos@node-0 ~]$ cd
+[centos@node-0 ~]$ cd example-app-frontend
+[centos@node-0 example-app-frontend]$ git checkout v2.0.0
+Previous HEAD position was fc0df8a fix warning
+HEAD is now at dc65f9b Change background color to green
+[centos@node-0 example-app-frontend]$
+```
+
+Now just build a new version of the frontend `v2.0.0` with the provided Dockerfile. And push the image to DockerHub. Use again your account.
+
+### Deploy frontend v2.0.0
+
+So now create again a YAML file `deployment_v2.yaml` for the new version and deploy it to the cluster. Don't forget to adapt the labels and the name. We don't want to perform an update on the exsiting deployment.
```

Comment:
```suggestion
Now create another YAML file `deployment_v2.yaml` for the new version and deploy it to the cluster. Don't forget to adapt the labels and the name. We don't want to perform an update on the existing deployment.
```

Reply:
the deployment/service manifest filenames are the same for backend and frontend. We should probably change it to something like `deployment_frontend_v2`